feat(footer): connected indicator added under chatbar (AP-356)

jpanay opened the pull request:
<!--  Thanks for sending a pull request!
If this is your first time, please read our contributor guidelines: https://github.com/Satellite-im/Core-PWA/wiki/Contributing
-->

**What this PR does** 📖
Shows if peer is connected or disconnected under chatbar
**Which issue(s) this PR fixes** 🔨
AP-356
<!--AP-X-->

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

phillsatellite commented:
Should indicator appear if user has no friends added?
<img width="1728" alt="Screen Shot 2022-02-04 at 2 26 59 PM" src="https://user-images.githubusercontent.com/93608357/152591082-305d466c-d81f-43ba-af78-d71dbdb622bb.png">

jpanay commented:
No it should not I'll fix that

On Fri, Feb 4, 2022 at 2:28 PM Phill Wisniewski ***@***.***>
wrote:

> Should indicator appear if user has no friends added?
> [image: Screen Shot 2022-02-04 at 2 26 59 PM]
> <https://user-images.githubusercontent.com/93608357/152591082-305d466c-d81f-43ba-af78-d71dbdb622bb.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Satellite-im/Core-PWA/pull/1330#issuecomment-1030286353>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQYRRHM3ZFHVU4C4XSXKP5DUZQSHPANCNFSM5NQYAQLQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

WanderingHogan commented on components/views/chat/chatbar/footer/Footer.vue:
```diff
@@ -1,5 +1,24 @@
 <template>
   <div id="bottom-bar-container">
+    <div v-if="!typing" class="is-connected">
```
Can you make this it's own ui chat component

jpanay replied:
All requests fixed, bug found by Phill fixed

WanderingHogan reviewed (changes requested):
Can you:
1. Split footer into a template, less, and vue file
2. Create a is-connected component
3. Add english to locales (in this repo, not the standalone repo)

WanderingHogan commented:
/rebase

stavares843 reviewed (changes requested):
https://user-images.githubusercontent.com/29093946/153056464-dfa50f04-0f32-4637-b21d-79865944f184.mov

says hogan is not connected on hogan's chat
change chat
still says hogan is not connected until page fully loads

stavares843 commented on components/views/chat/chatbar/footer/is-connected/Connected.html:
```diff
@@ -0,0 +1,22 @@
+  />
+  <TypographyText
+    :text="`is ${
+      $store.state.webrtc.connectedPeer ? 'connected' : 'not connected'
```
this should be added to locales the connected/not connected

stavares843 commented:
https://user-images.githubusercontent.com/29093946/153058038-12efeb58-f1e9-48d7-bb63-2897d627b6f2.mov


the connected/not connected updates fine

I closed the tab of user B and updated on user A saying it was no longer connected

but then returned error on user A
<img width="1175" alt="Captura de ecrã 2022-02-08, às 19 05 40" src="https://user-images.githubusercontent.com/29093946/153058120-e71a0adf-73b5-4b85-b538-1627e76f8a6f.png">

and after reloading the page says user B is connected again

<img width="1480" alt="Captura de ecrã 2022-02-08, às 19 08 34" src="https://user-images.githubusercontent.com/29093946/153058235-d603c577-7eb1-43ca-85d6-076b248ccde8.png">

but tab of user B was closed and not reopened

jpanay commented:
> Gravacao.do.ecra.2022-02-08.as.19.03.52.mov 
> the connected/not connected updates fine
> 
> I closed the tab of user B and updated user A saying it was no longer connected
> 
> but then returned error on user A <img alt="Captura de ecrã 2022-02-08, às 19 05 40" width="1175" src="https://user-images.githubusercontent.com/29093946/153058120-e71a0adf-73b5-4b85-b538-1627e76f8a6f.png">
> 
> and after reloading the page says user B is connected again
> 
> <img alt="Captura de ecrã 2022-02-08, às 19 08 34" width="1480" src="https://user-images.githubusercontent.com/29093946/153058235-d603c577-7eb1-43ca-85d6-076b248ccde8.png">
> 
> but the tab of user B was closed and not reopened

I'm trying to replicate this but I'm not getting that peer not connected error. I'm opening both tabs, waiting for connection, and closing one, but can't seem to get it to throw that error.

jpanay commented:
> > Gravacao.do.ecra.2022-02-08.as.19.03.52.mov
> > the connected/not connected updates fine
> > I closed the tab of user B and updated user A saying it was no longer connected
> > but then returned error on user A <img alt="Captura de ecrã 2022-02-08, às 19 05 40" width="1175" src="https://user-images.githubusercontent.com/29093946/153058120-e71a0adf-73b5-4b85-b538-1627e76f8a6f.png">
> > and after reloading the page says user B is connected again
> > <img alt="Captura de ecrã 2022-02-08, às 19 08 34" width="1480" src="https://user-images.githubusercontent.com/29093946/153058235-d603c577-7eb1-43ca-85d6-076b248ccde8.png">
> > but the tab of user B was closed and not reopened
> 
> I'm trying to replicate this but I'm not getting that peer not connected error. I'm opening both tabs, waiting for connection, and closing one, but can't seem to get it to throw that error.

Maybe I fixed it inadvertently with a few changes I made, can you test it again

phillsatellite commented:
I tested again, I got same error as @stavares843 after attempting to message user B after they disconnected, the only thing that was different on mine was when I reloaded page it still said User B was not connected 
<img width="1728" alt="Screen Shot 2022-02-10 at 11 47 00 AM" src="https://user-images.githubusercontent.com/93608357/153455959-a1dba613-5614-47cc-a9df-37eb12e809ff.png">

WanderingHogan commented:
@jpanay The error is here:
<img width="576" alt="image" src="https://user-images.githubusercontent.com/2993032/153531162-a5a65b76-72c9-478c-a5dd-678edd115a30.png">

typingNotifyHandler in the chatbar.vue file.

`const activePeer = this.$WebRTC.getPeer(activeFriend.address)` can come back as undefined

In lieue of digging into getPeer in the webrtc library, for now, please:

In the chatbar.vue file wrap the getPeer call in a try catch, 
<img width="599" alt="Screen Shot 2022-02-10 at 8 01 12 PM" src="https://user-images.githubusercontent.com/2993032/153531266-83865c38-e3f8-461a-b780-8e95e206c46e.png">

WanderingHogan commented:
@jpanay I can reproduce the error on my end

jpanay commented:
> @jpanay I can reproduce the error on my end

Yea I really don't know why I wasn't able to if all three of you can

stavares843 commented:
<img width="1512" alt="Captura de ecrã 2022-02-12, às 02 28 41" src="https://user-images.githubusercontent.com/29093946/153693108-956964ae-f0af-4e4f-8fe5-379e8bf6d6ca.png">


i have 2 users, they were both connected, I deactivated internet, and even now both have internet, still says one of them is not connected

stavares843 commented on components/views/chat/chatbar/footer/Footer.less:
```diff
@@ -0,0 +1,29 @@
+#bottom-bar-container {
+  display: flex;
+  justify-content: space-between;
+  // height: 1.5rem;
```
commented-out code 🔨

stavares843 commented:
/rebase

jpanay commented:
> <img alt="Captura de ecrã 2022-02-12, às 02 28 41" width="1512" src="https://user-images.githubusercontent.com/29093946/153693108-956964ae-f0af-4e4f-8fe5-379e8bf6d6ca.png">
> 
> i have 2 users, they were both connected, I deactivated internet, and even now both have internet, still says one of them is not connected
> 
> full flow was
> 
> * user A added user B
> * user B sent message to user A
> * user A sent message to user B
> * go to console network settings on user A and select offline
> 
> <img alt="Captura de ecrã 2022-02-12, às 02 30 12" width="660" src="https://user-images.githubusercontent.com/29093946/153693169-4b209aed-2a02-4ffd-b510-1b6403347e4c.png">
> 
> * user B sent message to user A - it sends
> * user A sent message to user B - doesn't send, there's no internet on that user
> * remove offline setting
> * user A sent message to user B - sends successfully
> * still shows user B not connected on user A side

Could you make a separate ticket for this, it's a webrtc issue beyond the scope of this PR

stavares843 commented:
the ticket which this tickets is expected to implement says the following:

''Show an indication under the chat input if the peer is connected/disconnected, might require checking/watching for events that don't exist?''

so, one of those events, would be a disconnected event as well, such as without internet

is not outside the scope, because the PR should include that doesn't matter if is or not a webrtc issue

but let me know cc @WanderingHogan

WanderingHogan commented:
I think that this is a problem with devtools in chrome - i dont think all connection types are getting closed when you disable the connection.

This acts differently when you are on two different devices. Eventually the webrtc connection times out and shows both users are offline. I think that there are problems managing the webrtc connection that should be included in future tickets but we need to debug some things first. 

If this ticket works otherwise and doesn't cause any crashes i think it can go in.
